Extend and improve versioning document (#333)
citynetwork/docs: docs/background/object-storage.md

For example, if you [set a public read policy](../howto/object-storage/s3/public-bucket.md) on an S3 bucket, the corresponding Swift container will still show an empty Read ACL, making the Swift container *look* like it is private, even though its objects are accessible through simple public URLs.
This is because Swift has no concept of fine-grained bucket policies, as they exist in S3.

## Object versioning

> In regular default buckets, each object has just one representation, and all operations (like removal or upload) operate on this one entity.

Reviewer: I would prefer something like this:

```suggestion
In regular default buckets, each object has a single representation.
All operations, such as removal or upload, may be performed on it.
```

> With versioned buckets, modifications (like uploading an object under the same name), only modify a view of the bucket.

Reviewer: ```suggestion
With versioned buckets, modifications like uploading an object with the same name only modify a view of the bucket.
```

> In this bucket operation mode, each object receives a unique version ID, and modifications retain prior versions.

Reviewer: ```suggestion
In this mode of operation, each object receives a unique version ID, and modifications retain prior versions.
```

| Operation                    | Default bucket                | Versioned bucket                                                      |
| ---------------------------- | ----------------------------  | -------------------------------------                                 |
| Uploading a new object       | New object is stored          | New object is stored                                                  |
| Overwrite of existing object | New object overwrites old one | New object version becomes current                                    |
| Delete object                | Object is deleted             | Delete marker is placed, subsequent object access returns "Not found" |

Once you [enable versioning on a bucket](../howto/object-storage/s3/versioning.md#enabling-bucket-versioning), you cannot turn it off --- you can, however, suspend it.
When versioning is suspended, new objects are created with a version ID of `null` and overwrite any existing `null` version, but all previously versioned objects remain unchanged and accessible.

### Delete markers

When you delete an object in a versioned bucket, the object isn't actually removed.

> Instead, a special placeholder called a delete marker acts as the current version of the object.

Reviewer: ```suggestion
Instead, a special placeholder named _delete&nbsp;marker_ acts as the current version of the object.
```
